src/text_agent.rs

```diff
@@ -0,0 +1,245 @@
+        let (sender, receiver) = crossbeam_channel::unbounded();
+        Self { sender, receiver }
+    }
+}
+
+pub fn propagate_text(
+    channel: Res<TextAgentChannel>,
+    mut context_params: ContextSystemParams,
+    mut redraw_event: EventWriter<RequestRedraw>,
+) {
+    for mut contexts in context_params.contexts.iter_mut() {
+        if contexts.egui_input.focused {
+            let mut redraw = false;
+            while let Ok(r) = channel.receiver.try_recv() {
+                redraw = true;
+                contexts.egui_input.events.push(r);
+            }
+            if redraw {
+                redraw_event.send(RequestRedraw);
+            }
+            break;
+        }
+    }
+}
+
+fn is_mobile() -> Option<bool> {
+    const MOBILE_DEVICE: [&str; 6] = ["Android", "iPhone", "iPad", "iPod", "webOS", "BlackBerry"];
+
+    let user_agent = web_sys::window()?.navigator().user_agent().ok()?;
+    let is_mobile = MOBILE_DEVICE.iter().any(|&name| user_agent.contains(name));
+    Some(is_mobile)
+}
+
+/// Text event handler,
+pub fn install_text_agent(sender: Sender<egui::Event>) -> Result<(), JsValue> {
+    let window = web_sys::window().unwrap();
+    let document = window.document().unwrap();
+    let body = document.body().expect("document should have a body");
+    let input = document
+        .create_element("input")?
+        .dyn_into::<web_sys::HtmlInputElement>()?;
+    let input = std::rc::Rc::new(input);
+    input.set_id(AGENT_ID);
+    let is_composing = Rc::new(Cell::new(false));
+    {
+        let style = input.style();
+        // Transparent
+        style.set_property("opacity", "0").unwrap();
+        // Hide under canvas
+        style.set_property("z-index", "-1").unwrap();
+
+        style.set_property("position", "absolute")?;
+        style.set_property("top", "0px")?;
+        style.set_property("left", "0px")?;
+    }
+    // Set size as small as possible, in case user may click on it.
+    input.set_size(1);
+    input.set_autofocus(true);
+    input.set_hidden(true);
+
+    if let Some(true) = is_mobile() {
+        // keydown
+        let sender_clone = sender.clone();
+        let closure = Closure::wrap(Box::new(move |event: web_sys::KeyboardEvent| {
+            if event.is_composing() || event.key_code() == 229 {
+                // https://www.fxsitecompat.dev/en-CA/docs/2018/keydown-and-keyup-events-are-now-fired-during-ime-composition/
+                return;
+            }
+            if "Backspace" == event.key() {
+                let _ = sender_clone.send(egui::Event::Key {
+                    key: egui::Key::Backspace,
+                    physical_key: None,
+                    pressed: true,
+                    modifiers: egui::Modifiers::NONE,
+                    repeat: false,
+                });
+            }
+        }) as Box<dyn FnMut(_)>);
+        document.add_event_listener_with_callback("keydown", closure.as_ref().unchecked_ref())?;
+        closure.forget();
```

Comment:
Instead of forgetting the closure, can we register it in the `SubscribedEvents` resource? (Feel free to move it to `lib.rs` as it resides in the `web_clipboard` module atm.)
This way users will be able to unsubscribe manually in case they'll want to have some sort of multi-app setup on their page.

Reply:
Addressed in https://github.com/mvlabat/bevy_egui/pull/279/commits/5985a6a596b258f36f442cd8a1201a17014ce6a3 I did multiple different `SubscribedEvents` due to some annoyances with types but could switch things.